index.rst

```diff
@@ -1,61 +1,311 @@
+
+  domain = salobj.Domain()
+
+  mtcs = MTCS(domain=domain)
+  comcam = ComCam(domain=domain)
+
+  asyncio.gather(mtcs.start_task, comcam.start_task)
+
+  # slew/track target
+  await mtcs.slew_icrs(ra=14., dec=-10., rot=0.)
+
+  # Select sources for processing in the background
+  select_sources_task = asyncio.create_task(
+    mtcs.rem.mtaos.cmd_selectSources.set_start(
+        ra=14.,
+        dec=-10.,
+        pa=0.,
+        type=WFSMode.ComCam
+      )
+    )
+
+  # piston camera hexapod
+  await mtcs.rem.hexapod_1.cmd_move.set_start(z=1000.)
+
+  # take first exposure
+  data_id_1 = await comcam.take_eng(exptime=30)
+
+  # piston hexapod
+  await mtcs.rem.hexapod_1.cmd_move.set_start(z=-1000.)
+
+  # tell MTAOS to start processing first image in the background
+  preproc_task_1 = asyncio.create_task(mtcs.rem.mtaos.cmd_preProcess.set_start(dataId=data_id_1))
+
+  # take second exposure
+  data_id_2 = await comcam.take_eng(exptime=30)
+
+  # tell MTAOS to start processing second image in the background
+  preproc_task_2 = asyncio.create_task(mtcs.rem.mtaos.cmd_preProcess.set_start(dataId=data_id_2))
+
+  # wait for background processes to finish
+  await asyncio.gather(select_sources_task, preproc_task_1, preproc_task_2)
+
+  # Process full array intra/extra focal images
+  await mtcs.rem.mtaos.cmd_runWEP.set_start(
+      intraId=data_id_1, extraId=data_id_2
+  )
+
+  # issue corrections to the components
+  await mtcs.rem.mtaos.cmd_issueCorrections.start()
+
+In the code above we used the already existing functionality of the `MTCS`_ and `ComCam`_ classes.
+Note that the process would be similar if we where to use ``LSSTCam`` instead of ``ComCam``, simply replacing the `ComCam`_ class by the equivalent when that becomes available.
+This means, SAL Scripts can be written for ``ComCam`` in a way that will be reusable for the main camera.
+
+.. _MTCS: https://ts-observatory-control.lsst.io/user-guide/maintel/mtcs-user-guide.html
+.. _ComCam: https://ts-observatory-control.lsst.io/user-guide/maintel/CCCamera-user-guide.html
+
+.. _user-provided-optical-aberration:
+
+User-provided optical aberration
+--------------------------------
+
+In some conditions is may be useful to allow users to specify optical aberrations to be introduced to the system, e.g., add a certain amount of comma to the system.
+The MTAOS can easily receive a list of Zernike coefficients, pass them through the Optical Feedback Control and compute the offsets and bending modes required.
+
+The command would simply receive an array of wavefront errors and apply them to the system properly registering the correction internally.
+
+.. code-block:: py
+
+  import numpy as np
+
+  # Assuming the wavefront correction is an array of 19 Zernike coefficients
+  wf = np.zeros(19)
+
+  # Assuming ANSI indexes, wf[7] is vertical comma
+  # Adding 1mm of vertical comma
+  wf[7] = 1.
+
+  await mtaos.cmd_runOFC.set_start(wf = wf)
```

Comment:
What is the meaning/responsibility of `runOFC` here? If it is to issue the correction to subsystems, why not just add an argument to the existed command such as `cmd_issueWavefrontCorrection`. In addition, you may also need to assign which CCD is for this `wf`. In addition, this use case is for single CCD or multiple CCDs. This information is important to design the interface of API.

Reply:
Good point. I think `wf` needs to be a dict, where the keys are the sensor names.
We will also need a `type` argument since ComCam and LSSTCam has different optics hence different sensitivity matrices.